languages/python/oso/oso/oso.py

```diff
@@ -45,17 +73,76 @@ def is_allowed(self, actor, action, resource) -> bool:
     def get_allowed_actions(self, actor, resource, allow_wildcard=False) -> list:
```

Comment:
Same comment about returning a Set instead of turning it back into a List

Reply:
Also as a side note: wherever you have `list` in a type position in Python I think you probably want to use `List[ElementType]` so you can provide a type for the element too

Reply:
Left this one returning a list -- I don't want to break existing behavior and we're deprecating this method anyway.

Changed the new methods to return `set`s 👍

Reply:
Good thinking 👍